kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/coordinatedcommits/InMemoryCommitCoordinator.scala

```diff
@@ -0,0 +1,366 @@
+     */
+    def lastRatifiedCommitVersion: Long = if (!active) -1 else maxCommitVersion
+
+    // Map from version to Commit data
+    val commitsMap: util.TreeMap[Long, Commit] = new util.TreeMap[Long, Commit]
+    // We maintain maxCommitVersion explicitly since commitsMap might be empty
+    // if all commits for a table have been backfilled.
+    val lock: ReentrantReadWriteLock = new ReentrantReadWriteLock()
+  }
+
+  private[coordinatedcommits] val perTableMap = new ConcurrentHashMap[String, PerTableData]()
+
+  override def registerTable(
+    logPath: Path,
+    currentVersion: Long,
+    currentMetadata: AbstractMetadata,
+    currentProtocol: AbstractProtocol): util.Map[String, String] = {
+    val newPerTableData = new PerTableData(currentVersion + 1)
+    perTableMap.compute(logPath.toString, (_, existingData) => {
+      if (existingData != null) {
+        if (existingData.lastRatifiedCommitVersion != -1) {
+          throw new IllegalStateException(
+            s"Table $logPath already exists in the commit-coordinator.")
+        }
+        // If lastRatifiedCommitVersion is -1 i.e. the commit-coordinator has never
+        // attempted any commit for this table => this table was just pre-registered. If
+        // there is another pre-registration request for an older version, we reject it and
+        // table can't go backward.
+        if (currentVersion < existingData.maxCommitVersion) {
+          throw new IllegalStateException(
+            s"Table $logPath already registered with commit-coordinator")
+        }
+      }
+      newPerTableData
+    })
+    Collections.emptyMap[String, String]()
+  }
+
+  override def commit(
+    logStore: LogStore,
+    hadoopConf: Configuration,
+    logPath: Path,
+    coordinatedCommitsTableConf: util.Map[String, String],
+    commitVersion: Long,
+    actions: util.Iterator[String],
+    updatedActions: UpdatedActions): CommitResponse = {
+    val tablePath = CoordinatedCommitsDefaultUtils.getTablePath(logPath)
+    if (commitVersion == 0) {
+      throw new CommitFailedException(false, false, "Commit version 0 must go via filesystem.")
+    }
+    logger.info("Attempting to commit version {} on table {}", commitVersion, tablePath)
+    val fs = logPath.getFileSystem(hadoopConf)
+    if (batchSize <= 1) {
+      // Backfill until `commitVersion - 1`
+      logger.info(
+        "Making sure commits are backfilled until {}" + " version for table {}",
+        commitVersion - 1,
+        tablePath)
+      backfillToVersion(
+        logStore, hadoopConf, logPath, coordinatedCommitsTableConf, commitVersion - 1, null)
+    }
+    // Write new commit file in _commits directory
+    val fileStatus = CoordinatedCommitsDefaultUtils.writeCommitFile(
+      logStore, hadoopConf, logPath.toString, commitVersion, actions, generateUUID)
+    // Do the actual commit
+    val commitTimestamp = updatedActions.getCommitInfo.getCommitTimestamp
+    val commitResponse = commitImpl(
```

Comment:
Can you invoke `addToMap` directly here? `commitImpl` does not seem to add anything.